v2/internal/frontend/desktop/windows/theme.go

```diff
@@ -7,32 +7,37 @@ import (
 
 func (w *Window) updateTheme() {
 
+	// Don't redraw theme if nothing has changed
+	if !w.themeChanged {
+		return
+	}
+	w.themeChanged = false
+
 	if win32.IsCurrentlyHighContrastMode() {
 		return
 	}
 
 	if !win32.SupportsThemes() {
 		return
 	}
-	// Only process if there's a theme change
-	isDarkMode := win32.IsCurrentlyDarkMode()
-	w.isDarkMode = isDarkMode
 
-	// Default use system theme
+	var isDarkMode bool
+	switch w.theme {
+	case windows.SystemDefault:
+		isDarkMode = win32.IsCurrentlyDarkMode()
+	case windows.Dark:
+		isDarkMode = true
+	case windows.Light:
+		isDarkMode = false
+	}
+	win32.SetTheme(w.Handle(), isDarkMode)
+
+	// Custom theme processing
 	winOptions := w.frontendOptions.Windows
 	var customTheme *windows.ThemeSettings
 	if winOptions != nil {
 		customTheme = winOptions.CustomTheme
-		if winOptions.Theme == windows.Dark {
-			isDarkMode = true
-		}
-		if winOptions.Theme == windows.Light {
-			isDarkMode = false
-		}
 	}
-
-	win32.SetTheme(w.Handle(), isDarkMode)
-
 	// Custom theme
 	if win32.SupportsCustomThemes() && customTheme != nil {
 		if w.isActive {
```

Comment:
Don't we need to also set `w.themeChanged = true` when `w.isActive` gets changed during `w32.WM_ACTIVATE`?
I think we might loose the adaption of the active/inactive color when customThemes are active.